internal/cmd/export.go

```diff
@@ -948,73 +960,313 @@ func gatherPostgresqlLogs(ctx context.Context,
 	cmd *cobra.Command,
 ) error {
 	writeInfo(cmd, "Collecting Postgres logs...")
-	// Get the primary instance Pod by its labels
-	pods, err := clientset.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{
-		// TODO(jmckulk): should we be getting replica logs?
-		LabelSelector: util.PrimaryInstanceLabels(clusterName),
+
+	dbPods, err := clientset.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{
+		LabelSelector: util.DBInstanceLabels(clusterName),
 	})
+
 	if err != nil {
 		if apierrors.IsForbidden(err) {
 			writeInfo(cmd, err.Error())
 			return nil
 		}
 		return err
 	}
-	if len(pods.Items) != 1 {
-		writeInfo(cmd, "No primary instance pod found for gathering logs")
+
+	if len(dbPods.Items) == 0 {
+		writeInfo(cmd, "No database instance pod found for gathering logs and config")
 		return nil
 	}
 
-	podExec, err := util.NewPodExecutor(config)
+	writeDebug(cmd, fmt.Sprintf("Found %d Pods\n", len(dbPods.Items)))
+
+	for _, pod := range dbPods.Items {
+		writeDebug(cmd, fmt.Sprintf("Pod Name is %s\n", pod.Name))
+
+		podExec, err := util.NewPodExecutor(config)
+		if err != nil {
+			return err
+		}
+
+		exec := func(stdin io.Reader, stdout, stderr io.Writer, command ...string,
+		) error {
+			return podExec(namespace, pod.Name, util.ContainerDatabase,
+				stdin, stdout, stderr, command...)
+		}
+
+		// Get Postgres Log Files
+		stdout, stderr, err := Executor(exec).listPGLogFiles(numLogs)
+		if err != nil {
+			if strings.Contains(stderr, "No such file or directory") {
+				writeDebug(cmd, "Cannot find any Postgres log files. This is acceptable in some configurations.\n")
+			}
+			if apierrors.IsForbidden(err) {
+				writeInfo(cmd, err.Error())
+				return nil
+			}
+			continue
+		}
+		if stderr != "" {
+			writeDebug(cmd, stderr)
+			continue
+		}
```

Comment:
```suggestion
		if err != nil {
			if apierrors.IsForbidden(err) {
				writeInfo(cmd, err.Error())
				return nil
			}
			continue
		}
		if stderr != "" {
			if strings.Contains(stderr, "No such file or directory") {
				writeDebug(cmd, "Cannot find any Postgres log files. This is acceptable in some configurations.\n")
			} else {
				writeDebug(cmd, stderr)
			}
			continue
		}
```
What do you think about moving the stderr check into its own discrete section?

Reply:
I think I can simplify this further if I can assume that any error would populate `err` and `stderr`. Is that a safe assumption in this case?

Reply:
Basically reduce that section to the following.

My overall goal is just to emit a helpful message when no PG logs are found, but still `continue` or return nil` when appropriate.

```
		// Get Postgres Log Files
		stdout, stderr, err := Executor(exec).listPGLogFiles(numLogs)
		if err != nil {
			if strings.Contains(stderr, "No such file or directory") {
				writeDebug(cmd, "Cannot find any Postgres log files. This is acceptable in some configurations.\n")
			}
			if apierrors.IsForbidden(err) {
				writeInfo(cmd, err.Error())
				return nil
			}
			writeDebug(cmd, stderr)
			continue
		}
```

Reply:
> I think I can simplify this further if I can assume that any error would populate err and stderr. Is that a safe assumption in this case?

I don't know if that's a safe assumption. The way I think about it, "stderr" is when the function returns an error, while "err" is the possibility that the function doesn't even run.

Some code for an executor: https://github.com/CrunchyData/postgres-operator-client/blob/9782c9b2d75a5cfbbc7992197bbd3199cb5437c9/internal/util/executor.go#L56-L64

For instance, if you run the command to find files, and there are no files, we get the stderr: "no files". But if you run the command to find files and Kubernetes has deleted your pod, we get an err telling us there's no pod with that name.